pkg/resource/template/resource.go

```diff
@@ -174,12 +189,23 @@ func (t *TemplateResource) createStageFile() error {
 // overwriting the target config file. Finally, sync will run a reload command
 // if set to have the application or service pick up the changes.
 // It returns an error if any.
-func (t *TemplateResource) sync() error {
+func (t *TemplateResource) sync() (ferr error) {
 	staged := t.StageFile.Name()
 	if t.keepStageFile {
 		log.Info("Keeping staged file: " + staged)
 	} else {
-		defer os.Remove(staged)
+		defer func() {
+			if e := os.Remove(staged); e != nil {
+				if !os.IsNotExist(e) {
```

Comment:
Shouldn't we only overwrite ferr if ferr is nil?

Actually, I'm not sure if this error should be propagated. We may have sync'd the file so returning err is prolly not buying us anything.

Reply:
yes on both your points. Let's just log the `os.Remove` error as `Error` so that they stand out in the logs.